tests/sla_tests.js

```diff
@@ -206,4 +252,51 @@ describe('SLA Bridge and Channel Tester', function() {
       }, asyncDelay);
     } 
   });
+  it('should enter the application but specify an invalid SLA bridge. The ' +
+      'inbound channel should be hung up before being answered by the app.',
+     function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound['inbound'] = 'yes';
+    var sla = require('../lib/sla.js')(client, inbound, 'invalid')
+      .catch(errHandler)
+      .done();
+
+    incorrectBridge();
+    function incorrectBridge() {
+      setTimeout(function() {
+        if(bridges.length === 0 && !dialed[0] && inbound['inbound'] &&
+          !inbound['answered'] && inbound['hungup'] && channels.length === 0) {
+            done();
+        } else {
+          incorrectBridge();
+        }
+      }, asyncDelay);
+    } 
+  });
+  it('should enter the application, call the dialed channel, but hang up ' + 
+      'before it answers (which in turn hangs up inbound caller)',
+     function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound['inbound'] = 'yes';
+    var sla = require('../lib/sla.js')(client, inbound, '999')
+      .catch(errHandler)
+      .done();
+    answeringDelay = 4 * asyncDelay;
+    inbound.hangup(function(){});
+
+    earlyHangup();
+    function earlyHangup() {
+      setTimeout(function() {
+        if(isMixing && channels.length === 0 && bridges.length === 1 &&
+          inbound['inbound'] && inbound['answered'] && inbound['hungup'] &&
```

Comment:
Since there's no _ in the property names, you can just use inbound.inbound, inbound.answered... Also, consider the names of boolean properties to clearly indicate their use. inbound.wasAnswered instead of inbound.answered.